packages/components/src/button/index.tsx

```diff
@@ -93,10 +106,12 @@ export function Button( props, ref ) {
 	);
 
 	const hasChildren =
-		children?.[ 0 ] &&
-		children[ 0 ] !== null &&
-		// Tooltip should not considered as a child
-		children?.[ 0 ]?.props?.className !== 'components-tooltip';
+		( 'string' === typeof children && !! children ) ||
+		( Array.isArray( children ) &&
+			children?.[ 0 ] &&
+			children[ 0 ] !== null &&
+			// Tooltip should not considered as a child
+			children?.[ 0 ]?.props?.className !== 'components-tooltip' );
```

Comment:
This is more complex than I'd like.

Simply checking for `Array.isArray( children ) && children?.[ 0 ]` wasn't enough, as `children` can be a `string`, which also has a `.length` property.